Kernel/Thread.cpp

```diff
@@ -884,14 +885,21 @@ const LogStream& operator<<(const LogStream& stream, const Thread& value)
     return stream << value.process().name() << "(" << value.pid() << ":" << value.tid() << ")";
 }
 
-void Thread::wait_on(WaitQueue& queue, Atomic<bool>* lock, Thread* beneficiary, const char* reason)
+Thread::BlockResult Thread::wait_on(WaitQueue& queue, timeval* timeout, Atomic<bool>* lock, Thread* beneficiary, const char* reason)
 {
     cli();
     bool did_unlock = unlock_process_if_locked();
     if (lock)
         *lock = false;
     set_state(State::Queued);
     queue.enqueue(*current);
+
+    u64 timer_id = 0;
+    if (timeout)
+        timer_id = TimerQueue::the().add_timer(*timeout, [&]() {
+            wake_from_queue();
+        });
```

Comment:
Please wrap multi-line consequent statements in `{ }` even if the compiler allows you not to.

Reply:
Ironically, I thought I was matching your style here, fixed. :+1: